pinot-core/src/main/java/com/linkedin/pinot/core/segment/SegmentNameConfig.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.segment;
+
+/**
+ * DO NOT remove any getters, setters, or fields from this class otherwise it will be backwards incompatible.
+ */
+public class SegmentNameConfig implements DefaultSegmentNameConfig {
+  private String _timeColumnName;
+  private String _tableName;
+  private String _segmentNamePostfix;
+  private String _segmentName;
+
+  public SegmentNameConfig() {
+
+  }
+  public SegmentNameConfig(String timeColumnName, String tableName, String segmentNamePostfix, String segmentName) {
```

Comment:
do not include segmentName in the constructor. Make it more difficult to use this by having a deprecated setter.